src/tools/compiletest/src/runtest.rs

```diff
@@ -3258,6 +3266,45 @@ impl<'test> TestCx<'test> {
         let stamp = crate::stamp(&self.config, self.testpaths, self.revision);
         fs::write(&stamp, compute_stamp_hash(&self.config)).unwrap();
     }
+
+    fn run_yk_tir_test(&self) {
+        let proc_res = self.compile_test();
+
+        if !proc_res.status.success() {
+            self.fatal_proc_rec("compilation failed!", &proc_res);
+        }
+
+        if !proc_res.status.success() {
+            self.fatal_proc_rec("test run failed!", &proc_res);
+        }
+
+        self.check_yk_tir_dump();
+    }
+
+    fn check_yk_tir_dump(&self) {
+        let test_file_contents = fs::read_to_string(&self.testpaths.file).unwrap();
+        if let Some(idx) = test_file_contents.find("// END RUST SOURCE") {
+            let (_, test_text) = test_file_contents.split_at(idx + "// END_RUST SOURCE".len());
+            let mut test_lines = vec![ExpectedLine::Elision];
+            for l in test_text.lines() {
+                if l.is_empty() {
+                    // ignore
```

Comment:
It might be easier to put `continue` in here. Or maybe `for l in test_text.lines().filter(|x| !x.is_empty())` is even clearer?

Reply:
Just following the precedent set in the `MirOpt` suite. This function was a copy-paste-modify kind of affair.

Reply:
Here's the code this was based on in rustc:
https://github.com/rust-lang/rust/blob/237bf3244fffef501cf37d4bda00e1fce3fcfb46/src/tools/compiletest/src/runtest.rs#L2944-L2946

Might as well follow upstream.